extensions/ecc/circuit/src/extension/mod.rs

```diff
@@ -17,15 +17,27 @@ use openvm_stark_backend::{
 };
 use serde::{Deserialize, Serialize};
 
-#[cfg(feature = "cuda")]
-mod cuda;
 mod weierstrass;
-
-#[cfg(feature = "cuda")]
-pub use cuda::*;
 pub use weierstrass::*;
 
+cfg_if::cfg_if! {
+    if #[cfg(feature = "cuda")] {
+        mod cuda;
+        pub use cuda::*;
+        pub use cuda::{
+            EccGpuProverExt as EccProverExt,
```

Comment:
I'm not sure external users really need to know about / use `ProverExt` at all fwiw, the builder is more important